### ISSUE TITLE
mdb-sql: Syntax error with negative floating point numbers

### ISSUE THREAD

**Issue by Bob620** [mdbtools/mdbtools#274]
I am attempting to use the mdb-sql tool in order to access mdb files on linux and mac. I have only tested with linux thus far.

When entering the SQL-subset of commands I need to run, I enter various lines such as:
```
SELECT * FROM Line WHERE 7.42040062 <= StageX AND 8.13515949 >= StageX AND -3.70763707 <= StageY AND -3.16938281 >= StageY
```

This returns the error:
```
Error at Line : syntax error near .70763707
syntax error near .70763707
Got no result for 'SELECT * FROM Line WHERE 7.42040062 <= StageX AND 8.13515949 >= StageX AND -3.70763707 <= StageY AND -3.16938281 >= StageY
' command
```

I run the same line on Windows and it works correctly, so it must be a problem with mdb-sql's implementation of negative floating point numbers.

My best guess (I can hardly read C) is that mdb-sql does not correctly recognize negative floating point numbers as valid floating point numbers, so it reads `-3.70763707` and gets `-3` as a number and is left with `.70763707` which is not valid syntax.

**Comment by evanmiller** [comment]
Please try the proposed fix in #279

**Comment by Bob620** [comment]
@evanmiller Works like a charm, thank you for the fast fix!
